### PR TITLE
Error flew non eu airline

### PR DESCRIPTION
**Pull request by aaschlote**
- Refactor: Add column "isEU" in the airline data source. For the "Sonstiges" airline, set as true, so it goes for the current error page
- Feat: Go to an error page in case the airline is not in EU
- Fix: In the Annullierung, when the start airport was outside EU, it was not going to kostenlos after select any airline

### CODE REVIEW

File: app/models/flows/fluggastrechte/guards.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import airlines from "data/airlines/data.json";
```
Comment (pgurusinga) [inline app/models/flows/fluggastrechte/guards.ts:1]: they are getting bigger now. We might need to think about whether we could do an async guard in the future.

Comment (pgurusinga) [reply app/models/flows/fluggastrechte/guards.ts:1]: I think server `app/models/flows/guards.server.ts`

File: app/models/flows/fluggastrechte/guards.ts [hunk 2]
```diff
@@ -125,14 +126,20 @@ export const guards = {
...
-    context.bereich === "nichtbefoerderung",
+    (context.bereich === "nichtbefoerderung" ||
+      context.bereich === "annullierung"),
+  isFluggesellschaftNoEU: ({ context }) => {
+    const isAirlineEU =
+      airlines.find((airline) => airline.iata === context.fluggesellschaft)
+        ?.isEU ?? false;
+
+    return !isAirlineEU;
+  },
```
Comment (pgurusinga) [inline app/models/flows/fluggastrechte/guards.ts:142]: can we add this to the test cases we have?

**Review by pgurusinga — APPROVED** [review]
beside the test cases of the Guards

File: app/models/flows/fluggastrechte/guards.ts (hunk 2 above)
Comment (aaschlote) [reply app/models/flows/fluggastrechte/guards.ts:142]: yeap, sure. I'll do it!

Comment (aaschlote) [reply app/models/flows/fluggastrechte/guards.ts:142]: Sure, I'll do it!

File: app/models/flows/fluggastrechte/guards.ts [hunk 3]
```diff
@@ -125,14 +126,20 @@ export const guards = {
...
   isKnownPartnerAirlineBereichVerspaetet: ({ context }) =>
     context.fluggesellschaft !== "sonstiges" &&
     context.bereich === "verspaetet",
-  isKnownPartnerAirlineBereichNichtBefoerderung: ({ context }) =>
+  isKnownPartnerAirlineBereichNichtBefoerderungOrAnnullierung: ({ context }) =>
     context.fluggesellschaft !== "sonstiges" &&
-    context.bereich === "nichtbefoerderung",
+    (context.bereich === "nichtbefoerderung" ||
+      context.bereich === "annullierung"),
+  isFluggesellschaftNoEU: ({ context }) => {
```
[the remark this answers is not in the document]
Comment (eliflores) [reply app/models/flows/fluggastrechte/guards.ts:136]: Created a PR with this suggested renaming: https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/928.
If it looks good, we can merge it into this PR ✨

File: app/models/flows/fluggastrechte/guards.ts [hunk 4]
```diff
@@ -125,14 +126,20 @@ export const guards = {
...
+    const isAirlineEU =
```
[the remark this answers is not in the document]
Comment (eliflores) [reply app/models/flows/fluggastrechte/guards.ts:137]: Created a PR with this suggested renaming: https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/928.
If it looks good, we can merge it into this PR ✨

File: scripts/airlines.ts [hunk 5]
```diff
@@ -4,13 +4,15 @@ import _ from "lodash";
...
 
 function processAirlineRow(row: string): Airline {
-  const columns = row.split(";");
+  const [airlineName, , iataCode, region] = row.split(";");
   return {
-    name: columns[0],
-    iata: columns[2],
+    name: airlineName,
+    iata: iataCode,
+    isEU: region === "EU",
```
[the remark this answers is not in the document]
Comment (eliflores) [reply scripts/airlines.ts:15]: Created a PR with this suggested renaming: https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/928.
If it looks good, we can merge it into this PR ✨

[the remark this answers is not in the document]
Comment (aaschlote) [reply scripts/airlines.ts:15]: Sounds better

File: app/models/flows/fluggastrechte/guards.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (aaschlote) [reply app/models/flows/fluggastrechte/guards.ts:136]: hmm, sounds good. I'll apply it to this PR

**Pull request merged**